cms/djangoapps/contentstore/xblock_storage_handlers/view_handlers.py

```diff
@@ -522,6 +528,59 @@ def create_item(request):
     return _create_block(request)
 
 
+def sync_library_content(downstream: XBlock, request, store) -> StaticFileNotices:
+    """
+    Handle syncing library content for given xblock depending on its upstream type.
+    It can sync unit containers and lower level xblocks.
+    """
+    link = UpstreamLink.get_for_block(downstream)
+    upstream_key = link.upstream_key
+    if isinstance(upstream_key, LibraryUsageLocatorV2):
+        lib_block = sync_from_upstream_block(downstream=downstream, user=request.user)
+        static_file_notices = import_static_assets_for_library_sync(downstream, lib_block, request)
+        store.update_item(downstream, request.user.id)
+    else:
+        with store.bulk_operations(downstream.usage_key.context_key):
+            upstream_children = sync_from_upstream_container(downstream=downstream, user=request.user)
+            downstream_children = downstream.get_children()
+            downstream_children_keys = [child.upstream for child in downstream_children]
+            # Sync the children:
+            notices = []
+            # Store final children keys to update order of components in unit
+            children = []
+            for i in range(len(upstream_children)):
+                upstream_child = upstream_children[i]
+                assert isinstance(upstream_child, LibraryXBlockMetadata)  # for now we only support units
+                if upstream_child.usage_key not in downstream_children_keys:
+                    # This upstream_child is new, create it.
+                    downstream_child = store.create_child(
+                        parent_usage_key=downstream.usage_key,
+                        position=i,
+                        user_id=request.user.id,
+                        block_type=upstream_child.usage_key.block_type,
+                        # TODO: Can we generate a unique but friendly block_id, perhaps using upstream block_id
+                        block_id=f"{upstream_child.usage_key.block_type}{uuid4().hex[:8]}",
+                        fields={
+                            "upstream": str(upstream_child.usage_key),
+                        },
+                    )
+                else:
+                    downstream_child_old_index = downstream_children_keys.index(upstream_child.usage_key)
+                    downstream_child = downstream_children[downstream_child_old_index]
+
+                result = sync_library_content(downstream=downstream_child, request=request, store=store)
+                children.append(downstream_child.usage_key)
+                notices.append(result)
+            for child in downstream_children:
+                if child.usage_key not in children:
+                    # This downstream block was added, or deleted from upstream block.
+                    store.delete_item(child.usage_key, user_id=request.user.id)
```

Comment:
Clarification: This is the part that (for the MVP) will delete any local additions to a Unit when the next upstream sync happens, right?

Reply:
Yes, this deletes two different things:
1) local additions in the course - will get deleted on next sync
2) blocks deleted upstream